runtime/wasm/src/module/mod.rs

```diff
@@ -161,7 +161,7 @@ impl WasmInstance {
         mut self,
         handler_name: &str,
     ) -> Result<BlockState, anyhow::Error> {
-        let block = self.instance_ctx().ctx.block.clone();
+        let block: &Arc<EthereumBlockType> = &self.take_ctx().ctx.block;
```

Comment:
Isn't this going to panic when `take_ctx()` is called again at the bottom of the method?